pdf/pdflinkarea.cpp

```diff
@@ -66,23 +73,40 @@ void PDFLinkArea::mousePressEvent(QMouseEvent *event)
 {
     d->clickInProgress = true;
     d->clickLocation = event->pos();
+    d->pressTimer.start();
 }
 
-void PDFLinkArea::mouseReleaseEvent(QMouseEvent *event)
+void PDFLinkArea::mouseMoveEvent(QMouseEvent *event)
 {
+    // Don't activate anything if the finger has moved too far
+    QRect rect((d->clickLocation - QPointF(d->wiggleFactor, d->wiggleFactor)).toPoint(),
+               QSize(d->wiggleFactor * 2, d->wiggleFactor * 2));
+    if (!rect.contains(event->pos()))
+        return;
     d->clickInProgress = false;
```

Comment:
clickInProgress was not introduced by me. I noticed that it's in fact unused but didn't touch it, not to interfere. I've now removed it completly. If you think I should not, I can put it back.